Source/Plugins/Core/com.equella.core/src/com/tle/core/auditlog/convert/AuditLogConverter.java

```diff
@@ -100,8 +104,22 @@ public void doExport(
 
     // write out the format details
     xmlHelper.writeExportFormatXmlFile(auditFolder, true);
-    AuditLogJavaDao.writeExport(
-        auditFolder, PER_XML_FILE, institution, message, xmlHelper, xstream);
+
+    int count = (int) auditLogDao.countByCriteria(Restrictions.eq("institution", institution));
+    if (count > 0) {
+      // Read all audit logs by paging.
+      for (int page = 0; page < count / PER_XML_FILE + 1; page++) {
+        List<AuditLogEntry> entries =
+            auditLogDao.findAllByCriteria(
+                null,
+                page * PER_XML_FILE,
+                PER_XML_FILE,
+                Restrictions.eq("institution", institution));
+
+        xmlHelper.writeXmlFile(auditFolder, page + ".xml", entries, xstream);
+        message.incrementCurrent();
+      }
```

Comment:
Could we go a little more declarative?

```suggestion
      IntStream.range(0,count / PER_XML_FILE + 1).forEach(page -> {
        List<AuditLogEntry> entries =
            auditLogDao.findAllByCriteria(
                null,
                page * PER_XML_FILE,
                PER_XML_FILE,
                Restrictions.eq("institution", institution));

        xmlHelper.writeXmlFile(auditFolder, page + ".xml", entries, xstream);
        message.incrementCurrent();
      }
```

Reply:
Not a problem!  I thought `IntStream` might be overkilled so I did not use it.

Reply:
On some level it is heavy handed, but seeing this loop is not the performance bottle neck in this block of code, I think the trade-off for robustness and readability is warranted.